src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4012,7 +4012,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
               reportAnnotationError(UnexpectedTreeAnnotationError(t, typedAnn))
           }
 
-          if (annType.typeSymbol == DeprecatedAttr && argss.flatten.size < 2)
+          if (annType.typeSymbol == DeprecatedAttr && argss.foldLeft(0)(_ + _.size) < 2)
```

Comment:
This is probably replacing temporary `::` allocations with temporary `Integer` box allocations, so I'd be inclined to leave it was as written. It doesn't matter for performance because we rarely get to this code.

Reply:
Ok. I have added a `sumSize` function that explicitly uses the `Int` base types, so as to avoid boxing, in the `Collections` file.